infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationStreamNameTableManager.java

```diff
@@ -161,8 +171,8 @@ private boolean tableVersionMatchesPlugin() {
             txn.commit();
         }
 
-        return record.getPayload() != null && (Objects.equals(record.getPayload().getVersion(),
-                logReplicationConfigAdapter.getVersion()));
+        return record.getPayload() != null &&
+                (record.getPayload().getVersion().equals(logReplicationConfigAdapter.getVersion()));
```

Comment:
this will throw an NPE if record.getPayload().getVersion() is null(possible on the very first startup of LR).

Reply:
We could add one more non-null check but I thought it would be a bit redundant. As long as `record` is not null (which implies the version string has been set), the `getVersion` will not return a null value. Moreover, the proto3 will give default values for the fields